### DISCUSSION TITLE
Cant change the value for percent impervious for a subcatchment

### DISCUSSION THREAD

**Discussion by Utsav1234567** [pyswmm/pyswmm#422] [General]
Q1. When I try to change one percent impervoius values, I cannot do it. 
(s1 = Subcatchments(sim)["S1"]
    print(s1.percent_impervious)
    s1.percent_impervious = 20
    print(s1.percent_impervious)
>10
>10)

Q2. Setting the percent imperviousness to random value does not work as it should either. I end up getting same total runoff flow from the catchment for all random numbers.

//////

# List to store results
results = []

# Loop for 10 simulations
for i in range(100):
    # Generate random percent impervious value between 10 and 50
    catchment_PI = np.random.uniform(10, 50)
    
    with Simulation(r'baseline.inp') as sim:
        Catchment = Subcatchments(sim)['C1']
        
        # Set catchment parameter
        Catchment.percent_imperviousness = catchment_PI
        
        time_stamps = []   
        sim.step_advance(1)

        # Launch a simulation!
        for ind, step in enumerate(sim):
            time_stamps.append(sim.current_time)

        # Obtain statistics from simulation  
        N_stat= Node.statistics 
        L=Link.depth   
        C_stat = Catchment.statistics
        
        # Append results to the list
        results.append((catchment_PI, C_stat['runoff']))

**Comment by karosc** [comment]
Looks like this functionality was planned but never implemented unfortunately. For now you should be able to use swmmio or swmm_api to open the inp file, change the fraction imperviousness, save a new inp file, then run pyswmm with that new file. If you're interested in contributing to the OWA SWMM repo to add this functionality, please do!

**Comment by Gerry-Xie** [comment]
Could author solve the problem that the impervious percentage of the sub-catchment cannot be modified?
